### PR TITLE
Make version upload process cancellable at the details step.

### LINKED ISSUE #4440: Adding a new listed version to an addon with unlisted versions forces dev to finish
Scenario: An add-on has one or more unlisted versions.  The developer starts to add a listed version and gets to the step after upload where they are required to provide some metadata (version license; add-on categories; add-on summary) to make the add-on complete enough to be sent for review.  The developer changes their mind about adding the version.  (Too bad!)

Once an add-on has a listed version attached to it we force the developer to supply the metadata required by redirecting all devhub pages under that add-on to the submission flow.  We do this because it's otherwise difficult to signal to the developer it's necessary to proceed - files/versions are created in AWAITING_REVIEW state (there is no Incomplete status for a file).  There is no cancel in the process - to regain normal add-on edit access the page must be filled in.

Possible solutions:
A) Nothing!  There are only 3 required items; summary is probably there already so it's a matter of clicking a random category and choosing a random license from the list.

B) Add a cancel button that would make the version STATUS_DISABLED and adjust the logic to only redirect when there is a non-disabled listed version.  Note this would mean the developer couldn't try again with the same version unless we keep the 'request review' function and make it handle this case.

C) Stop redirecting all the devhub pages into the flow.  This would likely mean a lot more confused developers who don't understand why the listed version they submitted isn't going to be reviewed even though we're telling them its awaiting review.  So we would need some UX changes in devhub with a clear call to action for the developer, both in the my submissions list, and in versions & status in the add-on page.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #4440 
Plus allow request review to restart the process with a disabled version.

### CODE REVIEW

File: src/olympia/devhub/templates/devhub/versions/list.html [hunk 1]
```diff
@@ -399,9 +399,8 @@ <h3 data-tmpl="{{ _('Delete Version {version}') }}"></h3>
       <h3>{{ _('Cancel Review Request') }}</h3>
       <p>
         {% trans %}
-            Canceling your review request will mark your add-on incomplete.
-            If you do not complete your add-on after several days
-            by re-requesting review, it will be deleted.
+            Canceling your review request will mark your add-on incomplete,
+            and any versions awaiting review for listing on AMO will be disabled.
```
Comment (jvillalobos) [inline src/olympia/devhub/templates/devhub/versions/list.html]: "for listing on AMO" seems redundant here

File: src/olympia/devhub/views.py [hunk 2]
```diff
@@ -1703,8 +1709,18 @@ def request_review(request, addon_id, addon):
...
+    if latest_version:
+        latest_version.files.filter(
+            status=amo.STATUS_DISABLED).update(
+            status=amo.STATUS_AWAITING_REVIEW)
+    if addon.has_complete_metadata():
+        addon.update(status=amo.STATUS_NOMINATED)
+        messages.success(request, _('Review Requested.'))
+    else:
+        messages.success(request, _(
+            'Review Requested.  You must provide further details to proceed.'))
```
Comment (jvillalobos) [inline src/olympia/devhub/views.py]: When would this message show up?

Comment (eviljeff) [inline src/olympia/devhub/views.py]: When you click 'request review' in devhub versions and status page - it shows up as a banner at the top of the details page.

**Comment by diox** [comment]
This looks good, but waiting on an answer on https://github.com/mozilla/addons-server/issues/4440#issuecomment-275547327 to finish reviewing

File: src/olympia/devhub/views.py (hunk 2 above)
Comment (jvillalobos) [inline src/olympia/devhub/views.py]: Ok. Please use lower case `r` for both `requested`.

**Pull request merged**